gossip/src/cluster_info.rs

```diff
@@ -2349,6 +2349,37 @@ pub struct NodeConfig {
     pub num_quic_endpoints: NonZeroUsize,
 }
 
+#[derive(Debug, Clone)]
+pub struct BindIpAddrs {
+    /// Primary IP address this node will bind to on startup
+    primary: IpAddr,
+    /// Secondary IP addresses
+    /// This node will bind to these addresses if the primary address is not available.
+    /// For some Solana protocols, this node will also bind to these addresses on startup
+    others: Vec<IpAddr>,
+}
+
+impl BindIpAddrs {
+    pub fn new(addrs: Vec<IpAddr>) -> Result<Self, String> {
```

Comment:
It may be wise to check that either 

- Exactly one address was provided (then it can be whatever)
- Multiple addresses were provided, and **none** of them are multicast, loopback or 0.0.0.0.

Holding this invariant will save us a bunch of heeadache in support. This can be trivially done since constructor is already fallible here.